app/models/concerns/user_encrypted_attribute_overrides.rb

```diff
@@ -1,25 +1,25 @@
 module UserEncryptedAttributeOverrides
   extend ActiveSupport::Concern
 
-  # override some Devise methods to support our use of encrypted_email
-
   class_methods do
-    def find_first_by_auth_conditions(tainted_conditions, opts = {})
-      if tainted_conditions[:email].present?
-        opts[:email_fingerprint] = create_fingerprint(tainted_conditions.delete(:email))
-      end
-      to_adapter.find_first(devise_parameter_filter.filter(tainted_conditions).merge(opts))
+    # override this Devise method to support our use of encrypted_email
+    def find_first_by_auth_conditions(tainted_conditions, _opts = {})
```

Comment:
since we are overriding a Devise method here, throwing away the `opts` and changing so much from the original logic smells dangerous to me. Are any future Devise plugins going to rely on `opts`? We don't know do we?